backend/rust/backend/src/main.rs

```diff
@@ -0,0 +1,334 @@
+        let reply = Reply {
+            message: "OK".into(),
+        };
+        let res = Response::new(reply);
+        Ok(res)
+    }
+
+    #[tracing::instrument]
+    async fn predict(&self, request: Request<PredictOptions>) -> Result<Response<Reply>, Status> {
+        // TODO: How to get model from load_model function?
+        todo!("How to get model from load_model function?")
+    }
+
+    #[tracing::instrument]
+    async fn load_model(
+        &self,
+        request: Request<ModelOptions>,
+    ) -> Result<Response<PbResult>, Status> {
+        let result = match request.get_ref().model.as_str() {
+            "mnist" => {
+                let model = MNINST::load_model(request.get_ref().clone());
+                let result= PbResult {
+                    message: format!("Model {} loaded successfully", request.get_ref().model),
+                    success: true,
+                };
+                Ok(Response::new(result))
+            }
+            _ => {
+                let result = PbResult {
+                    message: format!("Model {} not found", request.get_ref().model),
+                    success: false,
+                };
+                Ok(Response::new(result))
+            }
+        };
+        // TODO: add model to backend, how to transfer model to backend and let predict funciton can use it?
+        result
+    }
+
+    #[tracing::instrument]
+    async fn predict_stream(
+        &self,
+        request: Request<PredictOptions>,
+    ) -> Result<Response<ReceiverStream<Result<Reply, Status>>>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn embedding(
+        &self,
+        request: Request<PredictOptions>,
+    ) -> Result<Response<EmbeddingResult>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn generate_image(
+        &self,
+        request: Request<GenerateImageRequest>,
+    ) -> Result<Response<PbResult>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn audio_transcription(
+        &self,
+        request: Request<TranscriptRequest>,
+    ) -> Result<Response<TranscriptResult>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn tts(&self, request: Request<TtsRequest>) -> Result<Response<PbResult>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn tokenize_string(
+        &self,
+        request: Request<PredictOptions>,
+    ) -> Result<Response<TokenizationResponse>, Status> {
+        todo!()
+    }
+
+    #[tracing::instrument]
+    async fn status(
+        &self,
+        request: Request<HealthMessage>,
+    ) -> Result<Response<StatusResponse>, Status> {
+        // Here we do not need to cover the windows platform
+        let mut breakdown = HashMap::new();
+        let mut memory_usage: u64 = 0;
+
+        use nix::sys::resource::{getrusage, UsageWho};
+        let usage = getrusage(UsageWho::RUSAGE_SELF).expect("Failed to fet usage");
```

Comment:
@lu-zero Thank you very much. I have checked the `nix` repo and it is activate repo. So, I already replaced the code here.